### ISSUE TITLE
Screen reader testing

### ISSUE THREAD

**Issue by mgifford** [chinchang/hint.css#81]
Would be great to have some additinal screen reader testing.

The data-hint="LEFT-over Sir..."  isn't read at all in ChromeVox and is read no different than the rest of the text in VoiceOver.

I haven't tested in WindowsEyes, NVDA or JAWS yet.  

It would be important to have a pause of some sort at the very least. Alternately a prefix.  

I don't think there's any support for this, but something like http://www.w3.org/TR/css3-speech/ would be very useful.

**Comment by thierryk** [comment]
Accessibility is not limited to screen-readers. Using ARIA here won't help sighted keyboard users. Ironically I did a fiddle yesterday (updated: https://jsfiddle.net/thierrykoblentz/s1ht4j41/1/) to show how one could make the `title` attribute of `abbr` accessible to sighted users. That was a tongue in cheek example, but creating a tab stop (via `tabindex` for non focusable elements) may be the first step.

As a side note, regarding @mgifford comment:
```
[role="tooltip"] {
    background: none repeat scroll 0 0 #ef8100;
    display: none;
    margin: 0;
    z-index: 2;
}
```

The above should hide the tooltip from all users (including screen-reader users). How would that be revealed to keyboard users?

May be the easiest route to fix the screen-reader users experience would be to *not* hide the content of the tooltips in the first place, using a technique that make elements only invisible to sighted users: 
https://developer.yahoo.com/blogs/ydn/clip-hidden-content-better-accessibility-53456.html but that would require to have that content inside a node instead of the value of an attribute...

**Comment by jdan** [comment]
Hey all,

I've forked this repository and committed a proof-of-concept change to use `aria-label` as the source for the tooltip's contents: https://github.com/jdan/hint.css/commit/e6541166ec055c525485c5922027aab0b7a25004

You'll see that using `aria-label` instead of `data-hint` has the same results:

<img width="562" alt="screen shot 2016-02-01 at 10 24 25 am" src="https://cloud.githubusercontent.com/assets/287268/12722150/933ddab4-c8d1-11e5-8444-ee98785916cd.png">

But the tooltip's contents are now accessible to screen-readers:

<img width="819" alt="screen shot 2016-02-01 at 10 25 06 am" src="https://cloud.githubusercontent.com/assets/287268/12722175/af3892c2-c8d1-11e5-9ba9-0c23ca8a47ac.png">

**These changes follow the assumption that the tooltip is labeling the element it is annotating**. This means that screen-readers will not read off the actual content of the element itself, but only its tooltip. 

Consider the following:

![image](https://cloud.githubusercontent.com/assets/287268/12722301/4536a480-c8d2-11e5-9723-5febc877a05c.png)

In the above scenario, "Checkout" will be omitted in favor of "You will be charged $5.00" by the screen-reader. 

Happy to talk about this in detail! @chinchang / any other maintainers, let me know if these changes are desirable and I'll be happy to put up a PR.

**Comment by tfoxy** [comment]
I use the following css so I don't have to repeat the tooltip text twice:

```css
[data-hint][aria-label]:after {
    content: attr(aria-label);
}
[data-hint=''][aria-label]:before,
[data-hint=''][aria-label]:after {
    display: block !important;
}
[data-hint][aria-label='']:before,
[data-hint][aria-label='']:after {
    display: none !important;
}
```

```html
<a href="#" class="hint--bottom" data-hint aria-label="Some text"></a>
```

**Comment by chinchang** [comment]
@jdan This is nice! Thanks for setting up a fork. Now a way needs to be found out for getting this breaking change into the main library. IMO, the library should continue working with `data-hint` attribute but also support `aria-label` attribute so that existing apps don't break with new library. Summarizing the requirements:
1. If only `data-hint` present, use 'data-hint`
2. If only `aria-label` present, use `aria-label`
3. If both `data-hint` and `aria-label` present, use `aria-label`

Also, in your fork you can use `[class*="hint--"]` selector instead of using all `hint-*` classes to put base styles on...much smaller :)

I have created a WIP branch for accessibility - `a11y`. Would request you to please open a PR against that branch which can be taken forward.

@tfoxy I didn't get why you mentioned about 'repeating tooltip text twice'. :confused:

**Comment by tfoxy** [comment]
I was referring to the comment made by @Abbe98, where right now you have to put the same value for data-hint and aria-label if you don't add a css rule.

I like your suggestions. It would be great if there is an official support for aria-label.

**Comment by chinchang** [comment]
While we are on solving this, I had a doubt that I wanted to discuss here. I have heard a lot that a11y should be baked into your apps. I always researched a bit into analytics for things like screen readers etc and found that its hard to detect if your app is actually being used on a screen reader.

**My question is**: Why would I care about adding such a11y support (eg. screen reader support) when I don't really know if people really use my app with screen readers or not?

**Comment by mgifford** [comment]
Screen readers are just one type of disability. If you include people with hearing problems, mobility challenges, low vision, learning disabilities, as well as temporary conditions like injuries or incremental factors like aging, the % of folks with disabilities is closer to 20% of the population. 

So between that, increasing legislation and the moral arguments about rights to participation in society, there are a lot of reasons.  Do they speak to you and your project? Possibly not.

**Comment by chinchang** [comment]
🏁  This is finally live in v2.3.0! Thanks everyone who requested, helped and worked on this feature 😄
